rclpy/src/rclpy/node.cpp

```diff
@@ -140,19 +144,36 @@ get_node_names_impl(py::capsule pynode, bool get_enclaves)
     }
   }
 
-  rcutils_ret_t rcutils_ret;
-  rcutils_ret = rcutils_string_array_fini(&node_names);
-  if (RCUTILS_RET_OK != rcutils_ret) {
-    throw RCLError("Failed to destroy node names");
-  }
-  rcutils_ret = rcutils_string_array_fini(&node_namespaces);
-  if (RCUTILS_RET_OK != rcutils_ret) {
-    throw RCLError("Failed to destroy node_namespaces");
-  }
-  rcutils_ret = rcutils_string_array_fini(&enclaves);
-  if (RCUTILS_RET_OK != rcutils_ret) {
-    throw RCLError("Failed to destroy enclaves string array");
-  }
+  RCPPUTILS_SCOPE_EXIT(
+    {
+      fini_names_ret = rcutils_string_array_fini(&node_names);
+      fini_namespaces_ret = rcutils_string_array_fini(&node_namespaces);
+      fini_enclaves_ret = rcutils_string_array_fini(&enclaves);
+      if (RCUTILS_RET_OK != fini_names_ret) {
+        RCUTILS_SAFE_FWRITE_TO_STDERR(
+          "[rclpy|" RCUTILS_STRINGIFY(__FILE__) ":" RCUTILS_STRINGIFY(__LINE__) "]: "
+          "failed to fini node names during error handling: ");
+        RCUTILS_SAFE_FWRITE_TO_STDERR(rcl_get_error_string().str);
+        RCUTILS_SAFE_FWRITE_TO_STDERR("\n");
+        rcl_reset_error();
```

Comment:
@gbalke if an error was set by function A, resetting it before calling function B avoids warnings about error overwrite if function B happens to set an error as well. That is, consider moving each of these `if` clauses after the corresponding function call. It's pure printing, so no risk of an exception getting thrown there.

Reply:
I think I got mixed up because of https://github.com/ros2/rclpy/pull/730#discussion_r599111444. I'll change it back since I'm not throwing now.